doc/developers-guide/Hooks-and-handlers.md

```diff
@@ -28,21 +28,20 @@ I.e. if `mod_offline` was not available, the code would simply crash; if it was
 To avoid that coupling and also to enable other ([possibly yet to be written](#sidenote-yet-to-be-written)) code to carry out some action at this particular moment, `ejabberd_sm` instead calls:
 
 ```erlang
-ejabberd_hooks:run(offline_message_hook, LServer,
-                   [From, To, Packet])
+Acc1 = ejabberd_hooks:run_fold(offline_message_hook,
+                               LServer,
+                               Acc,
+                               [From, To, Packet])
 ```
 
 The extra level of indirection introduced by this call gives the flexibility to determine at runtime what code actually gets run at this point.
 `offline_message_hook` is just the name of the hook (in other words of the event that is being signalled);
 `From`, `To` and `Packet` are the arguments passed to the handler just as they would in case of the function being called directly;
 `LServer` is [the XMPP domain for which this hook is signalled](#sidenote-multiple-domains).
 
-
 ### Getting results from handlers
 
-Apart from being able to notify the rest of the system that some event has happened by running a hook with `ejabberd_hooks:run/3`, it's also possible to [fold](#sidenote-folds) over a sequence of handlers with `ejabberd_hooks:run_fold/4`.
-Like an ordinary `lists:foldl/3`, `ejabberd_hooks:run_fold/4` also requires an initial value to be passed to the function.
-
+Hook handlers are called by "folding", meaning iterating over the list of handlers passing them a set of arguments and some initial value which gets modified and returned by every handler.
```

Comment:
Hook handlers are called by "folding".
This means that each handler on a list is passed a set of arguments and an initial value that it then modifies, returns and hands over to the next handler in line.